pkg/startup/startup.go

```diff
@@ -1142,3 +1170,63 @@ func terminate() {
 	log.Warn("Terminating")
 	exitFunction(1)
 }
+
+func queryKubeadmConfigMap(clientset *kubernetes.Clientset) (*v1.ConfigMap, error) {
+	if clientset == nil {
+		return nil, fmt.Errorf("Not running on kubeadm")
+	}
+
+	cm, err := clientset.CoreV1().ConfigMaps(metav1.NamespaceSystem).Get(KubeadmConfigConfigMap, metav1.GetOptions{})
+	if err != nil {
+		// Any error other than not finding kubeadm's config map should be serious enough that
+		// we ought to stop here and return.
+		if !errors.IsNotFound(err) {
+			log.WithError(err).Error("failed to query kubeadm's config map")
+			terminate()
+			return nil, err
+		}
+	}
+
+	return cm, err
+}
+
+// extractKubeadmCIDRs looks through the config map and parses lines starting with 'podSubnet'.
+func extractKubeadmCIDRs(kubeadmConfig *v1.ConfigMap) (string, string, error) {
```

Comment:
I think we need UTs for this function, passing in a few different variations of the configmap and asserting we get the correct results.